### PR TITLE
Add documentation and sample applications for Automatic Trace ID Injection

### PR DESCRIPTION
**Pull request by zacharycmontoya**
Changes proposed in this pull request:
- Create a `demo` directory in the repo to highlight small code samples that our users can reference. I'm happy to rework the name as we all see fit.
- Add an AutomaticTraceIdInjection folder with a README that describes the additional change needed to get Trace/Log correlation working correctly, and sample apps for each supported logging framework with this capability:
  - log4net
  - NLog
  - Serilog

@DataDog/apm-dotnet

### CODE REVIEW

File: Datadog.Trace.sln [hunk 1]
```diff
@@ -262,6 +262,16 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "UpdateVendors", "tools\Upda
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SynchronizeInstaller", "tools\SynchronizeInstaller\SynchronizeInstaller.csproj", "{EBDDF6D7-59B8-4831-80C6-458C1ABBFC63}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "demo", "demo", "{34015B45-43F6-4E29-BDC6-4E16CFAD79BE}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "AutomaticTraceIdInjection", "AutomaticTraceIdInjection", "{5DE795C0-3DB6-4DD6-9EAF-F6A6698AD712}"
```
Comment (lucaspimentel) [inline Datadog.Trace.sln]: Thoughts on having these projects live in a separate solution? My VS already crawls loading 54 projects, and these are not needed for development.

Comment (zacharycmontoya) [reply Datadog.Trace.sln]: Yeah that's a good idea, 100% in support of that

Comment (bobuva) [reply Datadog.Trace.sln]: I agree too.

File: demo/AutomaticTraceIdInjection/SerilogExample/Program.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            File.Delete("log-Serilog-jsonFile-allProperties.log");
+            File.Delete("log-Serilog-compactJsonFile-allProperties.log");
+
+            var loggerConfiguration = new LoggerConfiguration()
+                                          .Enrich.FromLogContext()
+                                          .MinimumLevel.Is(Serilog.Events.LogEventLevel.Information);
+
+            // When using a message template, you must emit all properties in order to emit `dd.trace_id` and `dd.span_id`
+            // This is because Serilog cannot look up these individual keys by name due to the '.' in the key name (see https://github.com/serilog/serilog/wiki/Writing-Log-Events#message-template-syntax)
+            loggerConfiguration = loggerConfiguration
```
Comment (bobuva) [inline demo/AutomaticTraceIdInjection/SerilogExample/Program.cs]: 👍 on the research!

File: Datadog.Trace.sln (hunk 1 above)
Comment (zacharycmontoya) [reply Datadog.Trace.sln]: Resolved in eb1bce01779b16c74be7bd02e5ab13dac0abea61

**Comment by zacharycmontoya** [comment]
Thanks for the review! I'm going to wait to hear back from the logs team and validate before merging.

**Pull request merged**